pkg/controllers/chart/chart.go

```diff
@@ -831,3 +893,66 @@ func hashObjects(job *batch.Job, objs ...metav1.Object) {
 func setBackOffLimit(job *batch.Job, backOffLimit *int32) {
 	job.Spec.BackoffLimit = backOffLimit
 }
+
+func deepCopyStruct(source interface{}) interface{} {
```

Comment:
Why do we need custom deepcopy functions here? These are provided by Kubernetes for all the core types. See for example:
* https://pkg.go.dev/k8s.io/api/core/v1#SecurityContext.DeepCopy
* https://pkg.go.dev/k8s.io/api/core/v1#PodSecurityContext.DeepCopy